### ISSUE TITLE
[a11y] ARIA dialog and alertdialog nodes should have an accessible name (subplat)

### ISSUE THREAD

**Issue by pdehaan** [mozilla/fxa#12922]
## Description

> Ensures every ARIA `dialog` and `alertdialog` node has an accessible name ([ref](https://dequeuniversity.com/rules/axe/4.4/aria-dialog-name?application=AxeChrome))

### Steps to reproduce

1. Log into FxA and go to https://payments-stage.fxa.nonprod.cloudops.mozgcp.net/subscriptions
2. Have a pending cancelled subscription, then click the "Resubscribe" button.
3. Run axe DevTools

### Expected result

No a11y errors/warnings.

### Actual result

DOM element in question seems to be:
```html
<div class="portal" id="dialogs" role="dialog">
```

> Fix at least one (1) of these issues:
> - `aria-label` attribute does not exist or is empty
> - `aria-labelledby` attribute does not exist, references elements that do not exist or references elements that are empty
> - Element has no `title` attribute

Possibly add an `aria-label` of _"Resubscribe to { product-name }"_ or something.

### Environment

What browser and operating system are you using?

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-5117)
